experiments/AMIP/moist_mpi_earth/coupler_utils/bcfile_reader.jl

```diff
@@ -101,11 +109,11 @@ no_scaling(x, _info) = swap_space!(x, axes(_info.land_mask))
 
 # IO - monthly
 """
-    update_midmonth_data!(bcf_info)
+    update_midmonth_data!(FT, date, bcf_info)
 
 Extracts boundary condition data from regridded (to model grid) NetCDF files (which times, depends on the specifications in the `bcf_info` struct).
 """
-function update_midmonth_data!(date, bcf_info)
+function update_midmonth_data!(FT, date, bcf_info)
```

Comment:
```suggestion
function update_midmonth_data!(date, bcf_info)
```
And replace the deleted `FT = bcf_info.FT` with `FT = float_type(bcf_info)` below (and elsewhere in general)